File: 06/objc/parkinglot.m
#import "parkinglot.h"

@interface ParkingLot()

@end

@implementation ParkingLot

-(id) init {
	self = [super init];
	if(!self.vehicles) {
		self.vehicles = [[NSMutableArray alloc] init];	
	}
	return self;
}

- (void) addVehicle: (Vehicle*) newVehicle {
	[self.vehicles addObject: newVehicle];
}

- (void) printVehicles {
	for(int i = 0; i < [self.vehicles count]; i++) {
		[self.vehicles[i] printDetail];
	}
}

- (int) getVehicleCount {
	return [self.vehicles count];
}

- (void) hornEach {
	for(Vehicle* i in self.vehicles) {
		[i horn];
	}
}

@end

File: 06/parkinglot.cpp
#include "parkinglot.h"

using namespace std;

ParkingLot::ParkingLot() {
	//this->count_ = 0;
}

ParkingLot::~ParkingLot() {
	//delete this;
}

void ParkingLot::addVehicle(Vehicle* newVehicle) {
	this->vehicles_.push_back(newVehicle);
	//this->count_++;
}

void ParkingLot::printVehicles() {
	for(int i = 0; i < this->vehicles_.size(); i++) {
		this->vehicles_[i]->printDetail();
	}
}

int ParkingLot::getVehicleCount() {
	//return this->count_;
	return this->vehicles_.size();
}

void ParkingLot::hornEach() {
	for(int i = 0; i < this->vehicles_.size(); i++) {
		this->vehicles_[i]->horn();
	}
}
